numpy/_core/src/multiarray/item_selection.c

```diff
@@ -2877,64 +2943,13 @@ PyArray_Nonzero(PyArrayObject *self)
         npy_intp count = PyArray_DIM(self, 0);
         NPY_BEGIN_THREADS_DEF;
 
-        /* nothing to do */
-        if (nonzero_count == 0) {
-            goto finish;
-        }
-
         if (!needs_api) {
             NPY_BEGIN_THREADS_THRESHOLDED(count);
         }
 
         /* avoid function call for bool */
         if (is_bool) {
-            /*
-             * use fast memchr variant for sparse data, see gh-4370
-             * the fast bool count is followed by this sparse path is faster
-             * than combining the two loops, even for larger arrays
-             */
-            if (((double)nonzero_count / count) <= 0.1) {
-                npy_intp subsize;
-                npy_intp j = 0;
-                while (1) {
-                    npy_memchr(data + j * stride, 0, stride, count - j,
-                               &subsize, 1);
-                    j += subsize;
-                    if (j >= count) {
-                        break;
-                    }
-                    *multi_index++ = j++;
-                }
-            }
-            /*
-             * Fallback to a branchless strategy to avoid branch misprediction
-             * stalls that are very expensive on most modern processors.
-             */
-            else {
-                npy_intp *multi_index_end = multi_index + nonzero_count;
-                npy_intp j = 0;
-
-                /* Manually unroll for GCC and maybe other compilers */
-                while (multi_index + 4 < multi_index_end) {
-                    *multi_index = j;
-                    multi_index += data[0] != 0;
-                    *multi_index = j + 1;
-                    multi_index += data[stride] != 0;
-                    *multi_index = j + 2;
-                    multi_index += data[stride * 2] != 0;
-                    *multi_index = j + 3;
-                    multi_index += data[stride * 3] != 0;
-                    data += stride * 4;
-                    j += 4;
-                }
-
-                while (multi_index < multi_index_end) {
-                    *multi_index = j;
-                    multi_index += *data != 0;
-                    data += stride;
-                    ++j;
-                }
-            }
+            nonzero_idxs_1D_bool(count, nonzero_count, data, stride, multi_index);
```

Comment:
Doesn't this call belong to the family of fast functions in the other place?  I.e. only the manual loop of `nonzero` should stay here?

Reply:
I wanted to keep the diff short, but it indeed fits a bit better in the other block. Changed.